test/command_callback/test_cs_mcsc_command_callbacks.vader

```diff
@@ -0,0 +1,120 @@
+Before:
+  Save g:ale_cs_mcsc_options
+  Save g:ale_cs_mcsc_source
+  Save g:ale_cs_mcsc_assembly_path
+  Save g:ale_cs_mcsc_assemblies
+
+  unlet! g:ale_cs_mcsc_options
+  unlet! g:ale_cs_mcsc_source
+  unlet! g:ale_cs_mcsc_assembly_path
+  unlet! g:ale_cs_mcsc_assemblies
+
+  let g:temppath = fnamemodify(tempname(), ':p:h')
+  let g:temppathpattern = substitute(escape(b:temppath, '\\/.*$^~[]'), '[\\/]*$', '[\\\\/]\\+\\S\\+','')
+  let g:sometempfile = fnamemodify(b:temppath .'/some_temp_file.tmp', ':p')
+
+  runtime ale_linters/cs/mcsc.vim
+
+After:
+  Restore
+  unlet! b:ale_cs_mcsc_options
+  unlet! g:ale_cs_mcsc_source
+  unlet! g:ale_cs_mcsc_assembly_path
+  unlet! g:ale_cs_mcsc_assemblies
+  unlet! g:temppath
+  unlet! g:temppathpattern
+  unlte! g:sometempfile
+  call ale#linter#Reset()
+
+Execute(Check for proper default command):
+
+  let b:command = ale_linters#cs#mcsc#GetCommand(bufnr(''))
+  let b:command = substitute(b:command, '-out:' . g:temppathpattern, '-out:' . g:sometempfile, '')
+  let b:command = substitute(b:command, '\s\+', ' ', 'g')
```

Comment:
Add `unlet! b:command` to your `After` block above. Otherwise this variable might end up being used in some other test.